### PR TITLE
notebooks conversion

### PR DESCRIPTION
**Pull request by eaidova**
Uodate for noteboks conversion script.

* introduced skipping for notebooks required additional conditions (more memory, GPU e.t.c.), output of notebooks will be stored in the repo and conversion will use it instead execution on machine
* added timeout for killing run if notebook hangs
* better reporting of failures

TO DO:

- [ ]  make sure that all ignored notebooks have output
- [ ] fix failures

### CODE REVIEW

**Review by adrianboguszewski — COMMENTED** [review]
The only thing I would change is the name of the txt file. Maybe "exclude_notebooks_from_conversion.txt" is more meaningful as we have many ignore lists for different processes?

**Comment by adrianboguszewski** [comment]
And I also discovered locally the exclude_run.txt is ignored. 
image

File: .ci/convert_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                continue
+        else:
+            shutil.copyfile(notebook_path, notebook_executed)
+        markdown_retcode = subprocess.run(["jupyter", "nbconvert", "--to", "markdown", str(notebook_executed), "--output-dir", str(args.markdown_dir),
+                                          "--TagRemovePreprocessor.remove_all_outputs_tags=hide_output --TagRemovePreprocessor.enabled=True"], timeout=args.timeout).returncode
+
+        html_retcode = subprocess.run(["jupyter", "nbconvert", "--to", "html", str(notebook_executed), "--output-dir", str(args.html_dir),
+                                          "--TagRemovePreprocessor.remove_all_outputs_tags=hide_output --TagRemovePreprocessor.enabled=True"], timeout=args.timeout).returncode
+        rst_retcode = subprocess.run(["jupyter", "nbconvert", "--to", "rst", str(notebook_executed), "--output-dir", str(args.rst_dir),
+                                          "--TagRemovePreprocessor.remove_all_outputs_tags=hide_output --TagRemovePreprocessor.enabled=True"], timeout=args.timeout).returncode
```
Comment (adrianboguszewski) [inline .ci/convert_notebooks.py:61]: Shouldn't be under `if str(notebook_path) not in ignore_execution_list`?

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]

File: .ci/convert_notebooks.py (hunk 1 above)
Comment (eaidova) [reply .ci/convert_notebooks.py:61]: no, we have some notebooks which store output inside and we no need to spend time to execute them on ci (e.g. notebook requires GPU, but ci machine does not have it), so all actions except execution are acceptable for them, we just avoid execution

**Comment by eaidova** [comment]
> And I also discovered locally the exclude_run.txt is ignored. image

it is ok, in your logs started only conversion of notebook, but there is no line where notebook executed

**Comment by adrianboguszewski** [comment]
I see. What if we want to ignore some notebooks totally, like don't take any pre-generated output, just skip it?

**Comment by eaidova** [comment]
> I see. What if we want to ignore some notebooks totally, like don't take any pre-generated output, just skip it?

I can provide additional parameter for it, but could you please provide example such notebooks?

**Comment by adrianboguszewski** [comment]
I'm thinking about 108, which doesn't have included output, or any notebook that is hanging and also doesn't have included output. I don't say it should be a permanent solution. I just want to have a way to exclude some notebooks temporarily until we fix them as we did in CI checking.

File: .ci/ignore_convert_full.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+notebooks/108-gpu-device/108-gpu-device.ipynb
+notebooks/110-ct-segmentation-quantize/data-preparation-ct-scan.ipynb
+notebooks/110-ct-segmentation-quantize/pytorch-monai-training.ipynb
+notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb
```
[the remark this answers is not in the document]
Comment (adrianboguszewski) [reply .ci/ignore_convert_full.txt:4]: And 117 doesn't work for me at all. Shall we add it here?

**Pull request merged**